data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaTopicMetrics.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.util;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Map;
+import java.util.HashMap;
+
+public class KafkaTopicMetrics {
+    private long metricUpdateInterval;
+    private final String topicName;
+    private long updateTime;
+    private Map<KafkaConsumer, Map<String, Object>> consumerMetricsMap;
+    private Map<String, String> camelCaseMap;
+    private final PluginMetrics pluginMetrics;
+    
+    public KafkaTopicMetrics(final String topicName, final PluginMetrics pluginMetrics, final long metricUpdateInterval) {
+        this.pluginMetrics = pluginMetrics;
+        this.topicName = topicName;
+        this.consumerMetricsMap = new HashMap<>();
+        this.updateTime = Instant.now().getEpochSecond();
+        this.metricUpdateInterval = metricUpdateInterval;
+        this.camelCaseMap = new HashMap<>();
+        camelCaseMap.put("bytes-consumed-total", "bytesConsumedTotal");
+        camelCaseMap.put("records-consumed-total", "recordsConsumedTotal");
+        camelCaseMap.put("bytes-consumed-rate", "bytesConsumedRate");
+        camelCaseMap.put("records-consumed-rate", "recordsConsumedRate");
+        camelCaseMap.put("records-lag-max", "recordsLagMax");
+        camelCaseMap.put("records-lead-min", "recordsLeadMin");
+        camelCaseMap.put("commit-rate", "commitRate");
+        camelCaseMap.put("join-rate", "joinRate");
+        camelCaseMap.put("incoming-byte-rate", "incomingByteRate");
+        camelCaseMap.put("outgoing-byte-rate", "outgoingByteRate");
+        camelCaseMap.put("assigned-partitions", "outgoingByteRate");
```

Comment:
The value should be `assignedPartitions`. Maybe a copy-paste error?

Reply:
Actually, I don't need this. Thanks for finding this.